client/state/audio/middleware.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_RECEIVE_EVENT,
+} from 'state/action-types';
+
+const isAudioSupported = () => typeof window === 'object' && typeof window.Audio === 'function';
+
+export const playSound = src => {
+	if ( ! isAudioSupported() ) {
+		return;
+	}
+
+	const audioClip = new window.Audio( src );
+	audioClip.play();
+};
+
+export const onHappyChatMessage = ( dispatch, { event } ) => {
+	event && event.source !== 'customer' && playSound( '/calypso/audio/chat-pling.wav' );
+};
+
+/**
+ * Action Handlers
+ */
+
+// Initialized this way for performance reasons
+export const handlers = Object.create( null );
+handlers[ HAPPYCHAT_RECEIVE_EVENT ] = onHappyChatMessage;
+
+/**
+ * Middleware
+ */
+
+export default ( { dispatch, getState } ) => ( next ) => ( action ) => {
+	if ( ! isAudioSupported() ) {
+		return next( action );
+	}
```

Comment:
by putting this check here it will run on every action. however, since this is a curried function we can actually move it up a few levels and then it will only run once (in my example code I'm going to make it a named export so that we can expose it explicitly to tests)

```js
export const middleware = ( { dispatch } ) => next => {
	if ( ! isAudioSupported() ) {
		return next;
	}

	return next => {
		const handler = handlers[ action.type ];
		if ( handler ) {
			handler( dispatch, action );
		}

		return next( action );
	};
}
```